pageserver/src/tenant/ephemeral_file.rs

```diff
@@ -359,12 +409,14 @@ mod tests {
             assert_eq!(&buf, &content[i..i + 1]);
         }
 
-        let file_contents =
-            std::fs::read(file.buffered_writer.as_inner().as_inner().path()).unwrap();
-        assert_eq!(file_contents, &content[0..cap]);
+        let file_contents = std::fs::read(file.buffered_writer.as_inner().path()).unwrap();
+        assert!(file_contents == content[0..cap] || file_contents == content[0..cap * 2]);
```

Comment:
This assertion is now incorrect because the background write isn't guaranteed to
1. complete before we reach here
2. advance the file length atomically in TAIL_SZ units

Please invest some time to find a way to make the background flush timing deterministic & then make this test wait for the background flush to finish.

I think tokio's `start_paused=true` + `tokio::time::sleep(FOREVER)` will work - search for examples in the code base, or I can explain on a call.

Reply:
tried to do this using two oneshot channels: 5acc61bdbcda38c93e6825da91a18706d3f1a7f3